io/net/network_test.py

```diff
@@ -0,0 +1,247 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2016 IBM
+# Author: Prudhvi Miryala <mprudhvi@linux.vnet.ibm.com>
+# Author: Narasimhan V <sim@linux.vnet.ibm.com>
+#
+
+"""
+check the statistics of interface, test big ping
+test lro and gro and interface
+"""
+
+import time
+import hashlib
+import netifaces
+from avocado import main
+from avocado import Test
+from avocado.utils.software_manager import SoftwareManager
+from avocado.utils import process
+from avocado.utils import distro
+from avocado.utils import genio
+
+
+class NetworkTest(Test):
+    '''
+    To test different types of pings
+    '''
+
+    def setUp(self):
+        '''
+        To check and install dependencies for the test
+        '''
+        smm = SoftwareManager()
+        pkgs = ["ethtool", "net-tools"]
+        detected_distro = distro.detect()
+        if detected_distro.name == "Ubuntu":
+            pkgs.extend(["openssh-client", "iputils-ping"])
+        elif detected_distro.name == "SuSE":
+            pkgs.extend(["openssh", "iputils"])
+        else:
+            pkgs.extend(["openssh-clients", "iputils"])
+        for pkg in pkgs:
+            if not smm.check_installed(pkg) and not smm.install(pkg):
+                self.cancel("%s package is need to test" % pkg)
+        interfaces = netifaces.interfaces()
+        interface = self.params.get("interface")
+        if interface not in interfaces:
+            self.cancel("%s interface is not available" % interface)
+        self.iface = interface
+        self.peer = self.params.get("peer_ip")
+        if not self.peer:
+            self.cancel("No peer provided")
+        if not self.ping_check("-c 2"):
+            self.cancel("No connection to peer")
+        self.mtu = self.params.get("mtu", default=1500)
+
+    def test_mtu_set(self):
+        '''
+        set mtu size
+        '''
+        if not self.set_mtu_peer(self.mtu):
+            self.fail("Failed to set mtu in peer")
+        time.sleep(10)
+        if not self.set_mtu_host(self.mtu):
+            self.fail("Failed to set mtu in host")
+        time.sleep(10)
+
+    def test_gro(self):
+        '''
+        Test GRO
+        '''
+        self.receive_offload_toggle_test("gro")
+
+    def test_lro(self):
+        '''
+        Test LRO
+        '''
+        self.receive_offload_toggle_test("lro")
```

Comment:
LRO state change will fail for vNIC device... and LRO is fixed and user cannot change that.. can we handle that 

ethtool -K eth4 lro off
Cannot change large-receive-offload

Reply:
I can do that as enhancement.
The reason is, we need to cancel the test in that case, and that needs more work.
Works fine ?

Reply:
can we have a check like ?
if 'ibmvnic' not in  `ethtool -i eth1 | grep vnic`:

Reply:
# ethtool -k env7 | grep large-receive-offload
large-receive-offload: off [fixed]

Reply:
sriov:
ethtool -k eth8  | grep large-receive-offload
large-receive-offload: off